code/modules/antagonists/abductor/equipment/glands/access.dm

```diff
@@ -9,12 +9,4 @@
 
 /obj/item/organ/heart/gland/access/activate()
 	to_chat(owner, span_notice("You feel like a VIP for some reason."))
-	RegisterSignal(owner, COMSIG_MOB_ALLOWED, .proc/free_access)
-
-/obj/item/organ/heart/gland/access/proc/free_access(datum/source, obj/O)
-	SIGNAL_HANDLER
-	return TRUE
-
-/obj/item/organ/heart/gland/access/Remove(mob/living/carbon/M, special = 0)
-	UnregisterSignal(owner, COMSIG_MOB_ALLOWED)
-	..()
+	owner.AddComponent(/datum/component/simple_access, SSid_access.get_region_access_list(list(REGION_ALL_GLOBAL)), src)
```

Comment:
This component should be added to the gland on Initialize, not to the owner else he'll be stuck with permanent global access even after the gland is removed.

Reply:
The component is added to the mob, not the gland.

I can't put it on the gland initialize, because glands take some time to activate.

Reply:
Good catch, but we'll still have a problem with lingering global access unless the component is deleted when the gland is removed (you may want to store the reference in a variable instead of using component getters to avoid issues with the dupe mode).